flysystem to fetch calendars, contacts and profile for a user

ylebre opened the pull request.

ylebre committed:
WIP: flysystem to fetch calendars for a user

ylebre committed:
listing calendar

ylebre committed:
implementing 'has'

ylebre committed:
listing events

ylebre committed:
filled in file/item functions

ylebre committed:
typofix

ylebre committed:
also check for ics file

ylebre committed:
fix missing arg

ylebre committed:
add calendar item metadata

ylebre committed:
set mimetype for calendar items to text/calendar

ylebre committed:
add acl logic

ylebre committed:
move default acl to constructor

ylebre committed:
set default ACL

Potherca commented on src/NextcloudCalendar.php:
```diff
@@ -0,0 +1,443 @@
+
+
+/**
+ * Filesystem adapter to convert RDF files to and from a default format
```
Copy/Paste error?

ylebre replied:
fixed!

Potherca commented on src/NextcloudCalendar.php:
```diff
@@ -0,0 +1,443 @@
+    {
+        $this->userId = $userId;
+        $this->principalUri = "principals/users/" . $this->userId;
+	$this->defaultAcl = $defaultAcl;
```
You might want to check the tabs/spaces in this file.

ylebre replied:
fixed, converted all to spaces to comply with PSR

Potherca commented on src/NextcloudCalendar.php:
```diff
@@ -0,0 +1,443 @@
+        $logger = \OC::$server->getLogger();
+        $dispatcher = \OC::$server->getEventDispatcher();
+
+        $this->calDavBackend = new CalDavBackend($db, $principalBackend, $userManager, \OC::$server->getGroupManager(), $random, $logger, $dispatcher, true);	
```
For now this is fine but I think the `CalDavBackend` creation will eventually have to be moved outside of the class.

Potherca commented on src/NextcloudCalendar.php:
```diff
@@ -0,0 +1,443 @@
+     */
+    final public function readStream($path)
+    {
+        return false;
```
In other implementations where no stream is supported, we've just called `return $this->read($path);` here...
Is that a good idea?  :shrug:

ylebre replied:
agreed!

Potherca commented on src/NextcloudCalendar.php:
```diff
@@ -0,0 +1,443 @@
+     */
+    final public function write($path, $contents, Config $config)
+    {
+        $result = true;
```
This seems unused.

ylebre replied:
removed that line.

Potherca reviewed (approved):
I'm sad I didn't have the energy to build this, but your implementation seems spot on! Great work!

ylebre committed:
add profile flysystem adapter

ylebre committed:
change default filename to 'turtle'

ylebre committed:
typofixes

ylebre committed:
fix profile args

ylebre committed:
fix userId arg

ylebre committed:
drycoded contacts flysystem wrapper

ylebre committed:
Merge branch 'calendar-integration' of https://github.com/pdsinterop/flysystem-nextcloud into calendar-integration

ylebre committed:
init fixes

ylebre committed:
fix constructor

ylebre committed:
rename contacts to cards

ylebre committed:
tabs to spaces

ylebre committed:
cleanup fixes, remove unused code. call read/write in the stream counterparts.

ylebre committed:
fill read/write stream

ylebre committed:
fix comments, remove unused class

ylebre committed:
fix comment

Potherca committed:
Changes NextcloudCalendar based on review and add support for streams

- Adds missing items to doc-blocks
- Changes all quotes to the same type
- Fixes typo(s)
- Makes implicit declarations, returns and comparisons explicit
- Moves item check to normalize method
- Moves path splitting into a separate method
- Removes variable assignment in favor of directly returning a value
- Sort lists alphabetically

Potherca committed:
Change NextcloudContacts based on review, fix deleteDir function, and add support for streams

- Adds missing items to doc-blocks
- Changes all quotes to the same type
- Fixes typo(s)
- Makes implicit declarations, returns and comparisons explicit
- Moves item check to normalize method
- Moves path splitting into a separate method
- Removes variable assignment in favor of directly returning a value
- Sort lists alphabetically

Potherca committed:
Changes NextcloudProfile based on review and add support for streams

- Adds missing items to doc-blocks
- Changes all quotes to the same type
- Fixes typo(s)
- Makes implicit declarations, returns and comparisons explicit
- Removes variable assignment in favor of directly returning a value
- Sort lists alphabetically

Potherca commented:
@ylebre I've made the changes we discussed in person. Besides adding support for streams (using the polyfill provided by FlySystem) my changes also:

 - Add missing items to doc-blocks
 - Change all quotes to the same type
 - Fix typo(s)
 - Make implicit declarations, returns and comparisons explicit
 - Move item check to normalize method
 - Move path splitting into a separate method
 - Remove variable assignment in favor of directly returning a value
 - Sort lists alphabetically

Potherca committed:
Change short-syntax for to list() declaration.

Potherca committed:
Change profile path 'turtle' to 'card'.
